typewise_alert.py

```diff
@@ -1,46 +1,81 @@
 
-def infer_breach(value, lowerLimit, upperLimit):
-  if value < lowerLimit:
-    return 'TOO_LOW'
-  if value > upperLimit:
-    return 'TOO_HIGH'
-  return 'NORMAL'
-
-
-def classify_temperature_breach(coolingType, temperatureInC):
-  lowerLimit = 0
-  upperLimit = 0
-  if coolingType == 'PASSIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 35
-  elif coolingType == 'HI_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 45
-  elif coolingType == 'MED_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 40
-  return infer_breach(temperatureInC, lowerLimit, upperLimit)
-
-
-def check_and_alert(alertTarget, batteryChar, temperatureInC):
-  breachType =\
-    classify_temperature_breach(batteryChar['coolingType'], temperatureInC)
-  if alertTarget == 'TO_CONTROLLER':
-    send_to_controller(breachType)
-  elif alertTarget == 'TO_EMAIL':
-    send_to_email(breachType)
-
-
-def send_to_controller(breachType):
-  header = 0xfeed
-  print(f'{header}, {breachType}')
-
-
-def send_to_email(breachType):
-  recepient = "a.b@c.com"
-  if breachType == 'TOO_LOW':
-    print(f'To: {recepient}')
-    print('Hi, the temperature is too low')
-  elif breachType == 'TOO_HIGH':
-    print(f'To: {recepient}')
-    print('Hi, the temperature is too high')
+class TypewiseAlert:
+    def __init__(self, limits_for_types=None, alert_mail_details=None, alert_target_funcs=None):
+        # Default values can be moved out of code to DB as required
+        self.default_alert_funcs = \
+            {
+                'TO_CONTROLLER': self.send_controller_message,
+                'TO_EMAIL': self.send_email,
+                'TO_CONSOLE': self.send_to_console
+            }
+        self.default_limits_for_cooling_types = {
+            "PASSIVE_COOLING": (0, 35),
+            "MED_ACTIVE_COOLING": (0, 40),
+            "HI_ACTIVE_COOLING": (0, 45),
+        }
+        self.default_alert_mail_details = {
+            "TOO_LOW": {
+                "recipient": "low_temperature_breach_expert@bosch.com",
+                "email_message": "The temperature has dropped beyond lower breach limits. "
+                                 "Please take corrective action immediately."
+            },
+            "TOO_HIGH": {
+                "recipient": "high_temperature_breach_expert@bosch.com",
```

Comment:
Facility for different email recepients for low and high - is that intentional?

Reply:
Yes. Intentional, because I wanted it to scale with different types of breaches. High and low just being placeholders, here